src/pages/_app.js

```diff
@@ -33,8 +40,9 @@ export default function App({ Component, pageProps }) {
       <Component
         {...pageProps}
         onDelete={handleDelete}
-        transactions={transactions}
+        initialData={initialData}
         mode={mode}
+        onAddTransaction={handleAddTransaction}
         action={action}
         onCancel={handleCancel}
         onConfirmDelete={handleConfirmDelete}
```

Comment:
Here you should pass your handelMode() as onMode to solve the issues mentioned above.